add --inputFile <url..> option

PierreJeanjacquot opened the pull request:
`--inputFile <url..>` option for `test` and `run` commands

PierreJeanjacquot commented on cli/src/utils/prepareInputFile.js:
```diff
@@ -0,0 +1,32 @@
+export async function prepareInputFile(url) {
+  try {
+    /**
+     * the worker names the file after the url part after the last `/` occurrence 🤨 looks weird and leads to bugs.
```
internal discussion https://iexec-team-private.slack.com/archives/C07T6TDFXKN/p1733217622535559

cedric25 reviewed (commented):
How did you test it? How can I test it on my side?

Have you tested with 2 input files or just one?

How is the file getting accessed in the iApp? Should we create another project template with the appropriate code?

I'd suggest to add examples of both `test` and `run` commands, like:
```
iapp test --inputFile "https://cdn.prod.website-files.com/6646148828eddb19c172bf2a/665db3ba85a625628c353a64_Logo-RLC-Yellow.png"
```

Is that correct? Or it needs more single or double quotes?

PierreJeanjacquot commented on cli/README.md:
```diff
@@ -55,11 +55,17 @@ iexec decentralized application.
+Options:
+
+- use `--args <args>` to provide input arguments to your iApp during testing
+  (use quotes to provide multiple args).
```
`iapp test --args '"arg 1"' '"arg 2"'`won't work (`"arg 2"` will not be interpreted as a value for `--args`)

correct examples: 
- `iapp test --args '"arg 1" "arg 2"'`
- `iapp test --args "'arg 1' 'arg 2'"`
- `iapp test --args "\"arg 1\" \"arg 2\""`

PierreJeanjacquot replied:
added an example directly in the option description. 8edcdcc3630b645a6c9c5a7b438cd1b964998f27

PierreJeanjacquot commented on cli/README.md:
````diff
@@ -80,33 +86,50 @@ Description: Deploy your iApp on the iExec protocol in debug mode.
+
+```
+
+```
````
531c7c3e28cc568ddf3c2b599ee7291495dccf61

PierreJeanjacquot commented on cli/README.md:
```diff
@@ -80,33 +86,50 @@ Description: Deploy your iApp on the iExec protocol in debug mode.
+Example: iexec task show
+0x62ed16ebc52c9437af45f57dc30819254ce391633c090e125253726eb76e07b1 --download
+task-result.zip
+
```
merge automatic conflict resolution did some weirdness :cry: 
fixed 3b99ee26d0ced894f7cc17aeb00fb5c986d809bf

PierreJeanjacquot commented on cli/README.md:
````diff
@@ -55,11 +55,17 @@ iexec decentralized application.
 
 ```bash
-iapp test [--args <input>]
+iapp test [--args <input>] [--inputFile <url..>]
````
replaced `..` by `...`
`...` is used to specify the option is variadic, meaning it accepts an array of values or may be invoked multiple times in the same command.

Agreed that this notation is not ubiquitous, I added a full example in `--inputFile <url...>` option description to mitigate the confusion 9f078c7f27c20ecf092526d28cb233ab9fcfdf6f

PierreJeanjacquot commented:
@cedric25 Thank you for the review :pray: 
My apologies for the messed README, I didn't notice the refresh merge automatic fusion went bad :disappointed: 

I added detailed descriptions of `--args <args>` and `--inputFile <url...>` options to explain how to use them (directly in the CLI help).
I didn't report the full description in the README to avoid maintenance issues (my take for an exhaustive CLI documentation is to generate the documentation from the CLI help)

NB: detailed descriptions tend to bloat the terminal when printing help on 80 columns, I adjusted the print width to fit the user terminal width.

PierreJeanjacquot commented:
> ```
> iapp test --inputFile "https://cdn.prod.website-files.com/6646148828eddb19c172bf2a/665db3ba85a625628c353a64_Logo-RLC-Yellow.png"
> ```
> 
> Is that correct? Or it needs more single or double quotes?

quotes are not needed, the option is variadic multiple files can be added
- in the same option invocation `--inputFile https://foo.com/fileA.txt https://bar.io/fileB.json`
- by multiple invocations `--inputFile https://foo.com/fileA.txt --inputFile https://bar.io/fileB.json`
